app/src/main/java/com/danosoftware/galaxyforce/sprites/game/bases/explode/BaseMultiExploder.java

```diff
@@ -0,0 +1,174 @@
+                        ? base.halfHeight() : base.halfWidth()) * 1.5);
+                final int x = base.x() + (int) (radius * (float) Math.cos(angle));
+                final int y = base.y() - (int) (radius * (float) Math.sin(angle));
+                timedExplosions.add(
+                        new TimedExplosion(
+                                x,
+                                y,
+                                random.nextFloat() * MAXIMUM_EXPLOSION_START_TIME));
+            }
+
+            // pick a random timed-explosion and reset start time to 0.
+            // ensures at least one explosion starts immediately
+            int idx = random.nextInt(timedExplosions.size());
+            TimedExplosion timedExplosion = timedExplosions.get(idx);
+            timedExplosions.set(idx, new TimedExplosion(
+                    timedExplosion.getX(),
+                    timedExplosion.getY(),
+                    0f));
+        }
+    }
+
+    @Override
+    public ISpriteIdentifier getExplosion(float deltaTime) {
+        explosionTime += deltaTime;
+
+        // trigger any explosions that are due to start
+        Iterator<TimedExplosion> iterator = timedExplosions.iterator();
+        while (iterator.hasNext()) {
+            TimedExplosion timedExplosion = iterator.next();
+            if (explosionTime >= timedExplosion.explodeTime) {
+                iterator.remove();
+
+                final IBaseExplosion explosion = new BaseExplosion(
+                        timedExplosion.getX(),
+                        timedExplosion.getY(),
+                        sounds,
+                        vibrator);
+                explosion.startExplosion();
+                explodingBases.add(explosion);
+            }
+        }
+
+        Iterator<IBaseExplosion> explosionIterator = explodingBases.iterator();
+        while (explosionIterator.hasNext()) {
+            IBaseExplosion explosion = explosionIterator.next();
+            explosion.animate(deltaTime);
+            if (explosion.isFinished()) {
+                explosionIterator.remove();
+            }
+        }
+
+        // the main explosion will only start if all extra explosions have already started
+        if (timedExplosions.isEmpty() && !startedMainExplosion && explosionTime >= MAIN_EXPLOSION_START_TIME) {
+            startedMainExplosion = true;
+            sounds.play(SoundEffect.BIG_EXPLOSION);
+            vibrator.vibrate(VibrateTime.TINY);
+        }
+
+        //if (startedMainExplosion) {
+            mainExplosionTime += deltaTime;
+            return animation.getKeyFrame(mainExplosionTime, Animation.ANIMATION_NONLOOPING);
+        //}
+
+        // if we haven't started the main explosion, show the original frozen alien sprite
+        //return spriteToExplode;
+    }
+
+    @Override
+    public boolean finishedExploding() {
+        return animation.isAnimationComplete() && timedExplosions.isEmpty();
```

Comment:
We also need to check all started base explosions and confirm they have finished.